### PR TITLE
1.18.1

### PR DESCRIPTION
**Pull request by oxodao**

### CODE REVIEW

**Review by MaxNeedsSnacks — APPROVED** [review]

**Comment by oxodao** [comment]
Temporarly changed it as draft. For some reason no backups are generated on my server in 1.18. I will investigate this tomorrow, as I did not change much since my 1.17 PR and this worked for the past three months...

**Comment by oxodao** [comment]
After restarting the server and doing a manual backup it did work. I will let it run to be sure that it can also do auto ones after the delay and we'll see since I could not reproduce. If some people could test it and tell me if they encounter any issue that would be nice. Letting this as a draft for a few days to see if I can reproduce it in any way or if this was a one-time mystical issue.

EDIT: Ok `/ftbbackups start` works as intended but it seems like time based do not work. I'll take a look at it when I have time. Stays in draft.

**Comment by Tomatoh** [comment]
Yeah the issue seems to be that the ServerTick isn't firing?

**Comment by oxodao** [comment]
Finally got time to work on this again. Of course it was the ServerTick, the logicalsidedprovider now only gives access to the CLIENT WORLD and I autocompleted it without paying attention, thanks @Tomatoh :)

The PR is still not "READY" as I built a test version to use on my server with extra panic commands just in case the noSave is not back to false at some moment and it would need further debugging. Once I have tested it enough, I'll change this draft to a real PR again.

When this is done maybe I'll update it to 1.18.2 too if you are interested, but as there is really few mods updated yet I don't need it so I won't be doing it otherwise.

I dug a bit in the MC code and what happened to me should NOT have happened. If my guts are right, the fact that the game did not save was somehow a backup set the noSave to true, but when the server is shutting down (properly at least), the stopServer method will ensure all worlds are set back to `noSave = false`. So maybe my server crashed in the meantime of a backup right after I've made my build. That's my only explanation...


tl;dr: Found why the saving timer did not worked, fixed it. Added some debug command and will use a dev build on my private server to test this out. Once I'm confident enough this is working right I'll set this PR as `Ready for review` 

EDIT 2022-04-09: No new issues found since this message, so it looks like it's ready for review

**Comment by maddiemort** [comment]
It would be great to have this updated for 1.18.2 as well if that's possible! Looks like a lot of things are settling on 1.18.2 now.

**Comment by MichaelHillcox** [comment]
Hay, thank you for your contributions, and sorry for the slow response. 

As of now, we don't have any plans to bring FTB backups (1) to newer versions of Minecraft as we've switched over to FTB Backups (2).

If our stance changes, I'll be sure to reopen this request and merge it in.
